codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -215,25 +216,22 @@ public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf dat
             // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
 
-            verifyEndOfStreamNotReceived(stream);
             verifyGoAwayNotReceived();
-            verifyRstStreamNotReceived(stream);
 
             // We should ignore this frame if RST_STREAM was sent or if GO_AWAY was sent with a
             // lower stream ID.
-            boolean shouldApplyFlowControl = false;
             boolean shouldIgnore = shouldIgnoreFrame(stream, false);
```

Comment:
I think I'm leaning toward leaving the terminology as-is. The logic is not just around the listener call itself - there often is some minor state change involved, such as actually creating the stream upon receipt of HEADERS. So it does seem to be slightly more than `notifyListeners`.